src/main/java/org/junit/runners/Parameterized.java

```diff
@@ -276,10 +330,15 @@ private ParametersRunnerFactory getParametersRunnerFactory()
             }
         }
 
+        private Integer parameterCount;
```

Comment:
Having this set as a side-effect of calling `createRunners()` is fragile. It would be better to make this final and assign this at construction time. I added a commit that does ths.